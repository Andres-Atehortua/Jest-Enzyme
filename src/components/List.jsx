import React from "react";

const List = ({ title, list }) => {
  return (
    <section>
      <h2 className="big">{title}</h2>
      <ul>
        {list.map((item) => (
          <li key={item.id}>{item.name}</li>
        ))}
      </ul>
    </section>
  );
};

export default List;
