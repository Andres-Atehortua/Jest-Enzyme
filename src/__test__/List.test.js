import React from "react";
import List from "./../components/List";
import { configure, shallow /* mount, render*/ } from "enzyme";
import toJson from "enzyme-to-json"; // Dar otro formato al snapshot
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

describe("Pruebas a componente <List />", () => {
  const fruits = [
    { name: "Fresa", id: 1 },
    { name: "Uva", id: 2 },
    { name: "Mango", id: 3 },
    { name: "Naranja", id: 4 },
  ];
  let wrapper;
  // Antes de cada test se reasigna un nuevo wrapper.
  beforeEach(() => {
    wrapper = shallow(<List title="frutas" list={fruits} />);
  });

  test("Validar nodos", () => {
    // Validar si existe nodo
    expect(wrapper.find("h2").exists()).toBe(true);
    //Validar si tiene clase css
    expect(wrapper.find("h2").hasClass("big")).toBe(true);
    // Validar cantidad de elementos hijos
    expect(wrapper.find("ul").children().length).toBe(4);
    // Validar el contenido de texto del primer elemento
    expect(wrapper.find("li").first().text()).toBe("Fresa");
    // Validar el contenido de texto del ultimo elemento
    expect(wrapper.find("li").last().text()).toBe("Naranja");
    // Validar el contenido de texto del elemento cuyo indice especificamos
    expect(wrapper.find("ul").childAt(2).text()).toBe("Mango");
    // Validar tipo de nodo/elemento
    expect(wrapper.find("ul").childAt(1).type()).toBe("li");
    // Validar el contenido de html
    expect(wrapper.find("li").first().html()).toBe("<li>Fresa</li>");
  });

  test("Actualizar valizaciones en props", () => {
    // Validar cantidad de elementos li
    expect(wrapper.find("li").length).toBe(4);
    wrapper.setProps({ list: [{ name: "kiwi", id: 5 }] });
    expect(wrapper.find("li").length).toBe(1);
    wrapper.setProps({ title: "Super frutas" });
    expect(wrapper.find(".big").text()).toBe("Super frutas");
  });

  test("Validar que coincida con snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
