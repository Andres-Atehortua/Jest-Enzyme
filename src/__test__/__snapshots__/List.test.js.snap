// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a componente <List /> Validar que coincida con snapshot 1`] = `
<section>
  <h2
    className="big"
  >
    frutas
  </h2>
  <ul>
    <li
      key="1"
    >
      Fresa
    </li>
    <li
      key="2"
    >
      Uva
    </li>
    <li
      key="3"
    >
      Mango
    </li>
    <li
      key="4"
    >
      Naranja
    </li>
  </ul>
</section>
`;
